pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/UpsertTableSegmentPreloadIntegrationTest.java

```diff
@@ -211,7 +226,7 @@ public void testSegmentAssignment()
                 serverStarter.getConfig().getProperty(CommonConstants.Server.CONFIG_OF_INSTANCE_DATA_DIR);
             File[] files = new File(segmentDir, getTableName() + "_REALTIME").listFiles();
             for (File file : files) {
-              if (file.getName().contains("tmp") || file.getName().contains("consumer")) {
+              if (!file.getName().startsWith(getTableName())) {
```

Comment:
not sure i understand what this change means. maybe add a comment?

Reply:
aah, basically apart from the segment directories, pinot also creates multiple other directories. I simply wanted to exclude them as they don't contain immutable segments.